### ISSUE TITLE
filter expression with _or_

### ISSUE THREAD

**Issue by colinsullivan** [bram85/topydo#177]
Hello,

Currently tasks must match all elements in a filter expression (separated by spaces):

As described [here](https://www.topydo.org/#FilterExpressions):

> A filter expression consists of several terms separated by whitespace.
> A todo item must match each term in order to be selected/shown.

What if we could filter based on multiple expressions?  Perhaps a comma could be used, like this:

`topydo ls +MyProject,+MyOtherProject`

This would list tasks from either project.

**Comment by mruwek** [comment]
We've [discussed such feature](https://gitter.im/bram85/topydo?at=586a1bbfc02c1a3959da17b4) with @bram85 some time ago. It needs major changes in how expressions are evaluated. Seems like we would need to write real tokenizer and parser for it to work properly and with backward compatibility. I believe this won't happen before 0.13.

**Comment by mruwek** [comment]
@colinsullivan 
Also syntax from your example won't play well with something more complex:

`topydo ls +MyProject, +MyOtherProject +YetAnotherProject` - how it should be evaluated?

Is it: `+MyProject OR (+MyOtherPoject AND +YetAnotherProject)`
or maybe it should be: `+MyProject OR +MyOtherProject  AND +YetAnotherProject`?

**Comment by p-phillips** [comment]
My two-pennyworth - any parsing which assumes you understand the precedence of the operators is doomed to burn someone, sometime (me probably!).  Some sort of parenthesis should be allowed even if not mandatory, from a usability perspective.  As most of the time I suspect it will be used in config files for columns rather than for on-the-fly evaluations, it doesn't matter (to me anyway) if it is a bit clunky with quoted parentheses.

**Comment by bram85** [comment]
Agreed, it's on my personal wish list too. As @mruwek explained, it should be done properly by parsing expressions including parentheses, while still maintaining backward compatibility with the current expressions.

**Comment by TheCedarPrince** [comment]
Bump. Any progress on this feature? Still would be great to have. Thanks for a great tool!
